docs/rfc/rfc-027-p2p-message-bandwidth-report.md

```diff
@@ -0,0 +1,137 @@
+# RFC 27: P2P Message Bandwidth Report
+
+## Changelog
+
+- Nov 7, 2022: initial draft (@williambanfield)
+- Nov 15, 2022: draft completed (@williambanfield)
+
+## Abstract
+
+Node operators and application developers complain that Tendermint nodes consume
+larges amounts of network bandwidth. This RFC catalogues the major sources of bandwidth
+consumption within Tendermint and suggests modifications to Tendermint that may reduce
+bandwidth consumption for nodes.
+
+## Background
+
+Multiple teams running validators in production report that the validator consumes a lot of bandwidth.
+They report that a operators running on a network with hundreds of validators consumes multiple terabytes
+of bandwidth per day. Prometheus data collected from a validator node running on the Osmosis chain shows that Tendermint
+sends and receives large amounts of data to peers. In the nearly three hours of observation, Tendermint sent nearly 42 gigabytes
+and received about 26 gigabytes, for an estimated 366 gigabytes sent daily and 208 gigabytes received daily. While this is shy
+of the reported terabytes number, operators running multiple nodes for a 'sentry' pattern could easily send and receive a terabyte of data.
+
+Sending and receiving large amounts of data has a cost for node operators. Most cloud platforms charge for network traffic egress.
+Google Cloud charges between [$.05 to $.12 per gigabyte of egress traffic][gcloud-pricing], and ingress is free. Hetzner [charges 1€ per TB used over the 10-20TB base bandwidth per month][hetzner-pricing], which will be easily hit if multiple terabytes are sent and received per day. Using the values collected from the validator on Osmosis, a single node on Google cloud may cost $18 to $44 a day running on Google cloud. On Hetzner, the estimated 18TB a month of both sending and receiving may cost between 0 and 10 Euro a month per node.
+
+## Discussion
+
+### Overview of Major Bandwidth Usage
+
+To determine which components of Tendermint were consuming the most bandwidth, I gathered prometheus metrics from the [Blockpane][blockpane] validator running on the Osmosis network for several hours.
+The data reveal that three message types account for 98% of the total bandwidth consumed. These message types are as follows:
+
+1. [consensus.BlockPart][block-part-message]
+2. [mempool.Txs][mempool-txs-message]
+3. [consensus.Vote][vote-message]
+
+
+The image below of p2p data collected from the Blockpane validator illustrates the bandwidth consumption of these three message types in both data sending and receiving.
+
+![](images/total-message-send-rate.png)
+![](images/total-message-recv-rate.png)
```

Comment:
https://github.com/tendermint/tendermint/blob/wb/p2p-rfc/docs/rfc/rfc-027-p2p-message-bandwidth-report.md Doesn't seem to be displaying them

Reply:
Same here.

Reply:
Should be fixed now.